Content.Server/Github/GithubApiManager.cs

```diff
@@ -0,0 +1,320 @@
+        _cfg.OnValueChanged(CCVars.GithubMaxRetries, val => _maxRetries = val, true);
+        _cfg.OnValueChanged(CCVars.GithubRequestBuffer, _rateLimiter.UpdateRequestBuffer, true);
+
+        _sawmill = _log.GetSawmill("github");
+
+        TryInitializeApi();
+    }
+
+    #region Public functions
+
+    /// <summary>
+    /// Directly send a request to the API. This does not have any rate limits checks so be careful!
+    /// <b>Only use this if you have a very good reason to!</b>
+    /// </summary>
+    /// <remarks>If you want the safe way to send requests, look at <see cref="TryMakeRequestSafe"/>></remarks>
+    /// <param name="request">The request to make.</param>
+    /// <returns>The direct HTTP response from the API. If null the request could not be made.</returns>
+    public async Task<HttpResponseMessage?> TryMakeRequest(IGithubRequest request)
+    {
+        if (!ApiEnabled())
+        {
+            _sawmill.Info("Tried to make a github api request but the api was not enabled.");
+            return null;
+        }
+
+        var json = JsonSerializer.Serialize(request);
+        var payload = new StringContent(json, Encoding.UTF8, "application/json");
+
+        var builder = new UriBuilder(BaseUri);
+        builder.Port = -1;
+        builder.Path = request.GetLocation(_owner, _repository);
+
+        var httpRequest = new HttpRequestMessage
+        {
+            Method = request.RequestMethod,
+            Headers =
+            {
+                { AcceptHeader, AcceptHeaderType },
+                { AuthHeader, AuthHeaderBearer+_authToken },
+                { VersionHeader, VersionNumber },
+            },
+            RequestUri = builder.Uri,
+            Content = payload,
+        };
+
+        var response = await _http.Client.SendAsync(httpRequest);
+
+        // Update rate limit
+        var remainingRequests = TryGetLongHeader(response.Headers, RemainingHeader);
+        if (remainingRequests != null)
+            _rateLimiter.UpdateRequests(remainingRequests.Value);
+
+        _sawmill.Info($"Made a github api request to: {BaseUri+request.GetLocation(_owner, _repository)}");
+
+        return response;
+    }
+
+    /// <summary>
+    /// The standard way to make requests to the GitHub api. This will ensure that the request respects the rate limit
+    /// and will also retry the request if it fails. Awaiting this to finish could take a very long time depending
+    /// on what exactly is going on! Only await for it if you're willing to wait a long time.
+    /// </summary>
+    /// <param name="request">The request you want to make.</param>
+    /// <param name="attempts">The number of attempts so far. Probably don't set this yourself!</param>
+    /// <returns>The direct HTTP response from the API. If null the request could not be made.</returns>
+    public async Task<HttpResponseMessage?> TryMakeRequestSafe(IGithubRequest request, uint attempts = 0)
```

Comment:
Just make this private, and add a public override that calls that with `attempts = 0` to hide that parameter. I can't think of any reason for an API user to inject a different attempts amount here.